include/graaflib/algorithm/coloring/welsh_powell.tpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <graaflib/algorithm/coloring/welsh_powell.h>
+
+#include <unordered_map>
+#include <vector>
+#include <algorithm>
+#include <iostream>
+
+namespace graaf::algorithm {
+
+template <typename GRAPH>
+std::unordered_map<vertex_id_t, int> welsh_powell_coloring(const GRAPH& graph) {
+    
+    using degree_vertex_pair = std::pair<int, vertex_id_t>;
+
+    // Step 1: Sort vertices by degree in descending order
+    std::vector<degree_vertex_pair> degree_vertex_pairs;
+    for (const auto& vertex : graph.get_vertices()) {
+        int degree = graph.get_degree(vertex);
+        degree_vertex_pairs.emplace_back(degree, vertex);
+    }
+
+    std::sort(degree_vertex_pairs.rbegin(), degree_vertex_pairs.rend());
+
+    // Step 2: Assign colors to vertices
+    std::unordered_map<vertex_id_t, int> color_map;
+
+    for (const auto [_, current_vertex] : degree_vertex_pairs) {
+        vertex_id_t current_vertex = degree_vertex_pair.second;
```

Comment:
I think this is why the CI is currently failing. Now that we unpack the pair we should use `current_vertex` directly rather than `degree_vertex_pair` (which is not defined in this scope).

```suggestion
    for (const auto [_, current_vertex] : degree_vertex_pairs) {
```

Reply:
Done!!!